nemo/lightning/pytorch/strategies.py

```diff
@@ -628,17 +637,39 @@ def load_checkpoint(self, checkpoint_path: Union[str, Path]) -> Dict[str, Any]:
         sharded_state_dict = {}
         sharded_state_dict["state_dict"] = self.megatron_parallel.sharded_state_dict()
 
-        if self.ckpt_include_optimizer and self.trainer.state.fn == TrainerFn.FITTING:
+        if (self.ckpt_include_optimizer or include_optimizer_states) and self.trainer.state.fn == TrainerFn.FITTING:
```

Comment:
I feel like it's a bit confusing to have two different arguments with nearly the same name. What about instead we add `ckpt_load_optimizer` and `ckpt_save_optimizer` attributes to the strategy?

Reply:
Yeah I think that should work. I'm ok with that.

Reply:
Good point. What about just renaming `include_optimizer_states` to `load_optimizer_states_from_ckpt` or something similar?